pkg/services/dashboards/service/dashboard_service.go

```diff
@@ -1620,6 +1621,15 @@ func (dr *DashboardServiceImpl) saveDashboardThroughK8s(ctx context.Context, cmd
 		return nil, err
 	}
 
+	// Grant permissions for items created at the root folder
+	if cmd.FolderUID == "" {
```

Comment:
just a note that this function won't be called when going through /apis directly

Reply:
As discussed, we will leave here and add the annotation to manifest on the UI